rviz_default_plugins/src/rviz_default_plugins/displays/image/ros_image_texture.cpp

```diff
@@ -250,6 +252,92 @@ bool ROSImageTexture::fillWithCurrentImage(sensor_msgs::msg::Image::ConstSharedP
   return new_image_;
 }
 
+struct yuyv
+{
+  uint8_t y0;
+  uint8_t u;
+  uint8_t y1;
+  uint8_t v;
+};
+
+struct uyvy
+{
+  uint8_t u;
+  uint8_t y0;
+  uint8_t v;
+  uint8_t y1;
+};
+
+// Function converts src_img from yuv to rgb
+void ROSImageTexture::imageConvertYUV422ToRGB(
+  uint8_t * dst_img, uint8_t * src_img,
+  int dst_start_row, int dst_end_row,
+  int dst_num_cols, uint32_t stride_in_bytes,
+  std::string src_format)
+{
+  int final_y0 = 0;
+  int final_u = 0;
+  int final_y1 = 0;
+  int final_v = 0;
+  int r1 = 0;
+  int b1 = 0;
+  int g1 = 0;
+  int r2 = 0;
+  int b2 = 0;
+  int g2 = 0;
+
+  uint32_t stride_in_pixels = stride_in_bytes / 4;
+
+  // rows in dst_img
+  for (int row = dst_start_row; row < dst_end_row; row++) {
+    // col iterates till num_cols / 2 since two rgb pixels processed each
+    // iteration cols in dst_img
+    for (int col = 0; col < dst_num_cols / 2; col++) {
+      if (!src_format.compare(sensor_msgs::image_encodings::YUV422_YUY2)) {
```

Comment:
This seems expensive to do in the "hot" part of the loop.  Since this doesn't change, at the very least you could do this string comparison before the loops start, and then just reference an integer.  Even better would be to hoist this branch out of the loops completely, which you could probably do by splitting this into two separate functions (one for YUV422 and one for YUV422_YUY2).

Reply:
Split into 2 seperate function for YUV422 and YUV422_YUY2 format. Check initially made for which one to use inside `setFormatAndNormalizeDataIfNecessary`